src/vs/workbench/parts/tasks/common/problemMatcher.ts

```diff
@@ -179,16 +178,16 @@ export interface ProblemMatch {
 }
 
 export interface HandleResult {
-	match: ProblemMatch;
+	match: ProblemMatch | null;
 	continue: boolean;
 }
 
 export function getResource(filename: string, matcher: ProblemMatcher): URI {
 	let kind = matcher.fileLocation;
-	let fullPath: string;
+	let fullPath: string = '';
 	if (kind === FileLocationKind.Absolute) {
 		fullPath = filename;
-	} else if (kind === FileLocationKind.Relative) {
+	} else if ((kind === FileLocationKind.Relative) && matcher.filePrefix) {
 		fullPath = Paths.join(matcher.filePrefix, filename);
 	}
 	fullPath = fullPath.replace(/\\/g, '/');
```

Comment:
Although this shouldn't happen I would code that differntly to ensure we see exception and not let a broken URI escape:

let fullPath: string | undefined;
...
if (fullPath === void 0) {
	throw new Error('This should never happen.');
}